pkg/client/unversioned/request.go

```diff
@@ -642,6 +642,7 @@ func (r *Request) Watch() (watch.Interface, error) {
 		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
+		defer resp.Body.Close()
```

Comment:
My bad, my understanding of `Close()` was wrong, we should close the body in this function. Although shall we do it out of the if condition, right after the error check?